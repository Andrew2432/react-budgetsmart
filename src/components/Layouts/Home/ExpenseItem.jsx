import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import { CollectionItem, Icon } from 'react-materialize';
import ExpenseContext from '../../../context/Expense/expenseContext';
import ConfirmModal from './ConfirmModal';

const ExpenseItem = (props) => {
  const expenseContext = useContext(ExpenseContext);
  const { editExpense, deleteExpense } = expenseContext;
  const { id, name, cost, createdAt } = props;

  const findID = (e) =>
    e.target.parentElement.parentElement.attributes.id.nodeValue;

  const handleEdit = (e) => {
    e.preventDefault();
    editExpense(findID(e));
  };

  const handleDelete = (id) => {
    // e.preventDefault();
    console.log(id);
    deleteExpense(id);
  };

  return (
    <CollectionItem id={id} data-key={id}>
      <b>{name}</b> | <em>Rs. {cost}</em> <br />
      <b>{createdAt}</b>
      <a className="secondary-content modal-trigger" href="#modal-1">
        <ConfirmModal
          id="modal-1"
          message="Are you sure you want to delete?"
          onClick={deleteExpense}
          targetid={id}
        />
        <Icon className="red-text">delete</Icon>
      </a>
      <a className="secondary-content" href="!#" onClick={handleEdit}>
        <Icon className="orange-text">mode_edit</Icon>
      </a>
    </CollectionItem>
  );
};

ExpenseItem.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  cost: PropTypes.number.isRequired,
  createdAt: PropTypes.string.isRequired,
};

export default ExpenseItem;
