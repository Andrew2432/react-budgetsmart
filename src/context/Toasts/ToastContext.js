import { createContext } from 'react';

// Create a context for toasts and export it
const ToastContext = createContext();

export default ToastContext;
