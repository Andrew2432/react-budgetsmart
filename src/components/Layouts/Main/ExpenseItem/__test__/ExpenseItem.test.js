import TestRenderer from 'react-test-renderer';
import React from 'react';
import ExpenseItem from '../ExpenseItem';
import ToastContext from '../../../../../context/Toasts/ToastContext';
import ExpenseContext from '../../../../../context/Expense/expenseContext';

describe('AddExpense component', () => {
  test('Renders correctly', () => {
    const expenseItem = new TestRenderer.create(
      (
        <ExpenseContext.Provider value={{ expenses: [] }}>
          <ToastContext.Provider value={{ messages: [] }}>
            <ExpenseItem />
          </ToastContext.Provider>
        </ExpenseContext.Provider>
      )
    );
    expect(expenseItem).toMatchSnapshot();
  });
});
