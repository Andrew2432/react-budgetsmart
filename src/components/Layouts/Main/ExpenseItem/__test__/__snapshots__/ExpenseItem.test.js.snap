// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddExpense component Renders correctly 1`] = `
<li
  className="collection-item"
>
  <b />
   | 
  <em>
    Rs. 
  </em>
   
  <br />
  <b />
  <a
    className="secondary-content"
    href="!#"
    onClick={[Function]}
  >
    <i
      className="material-icons red-text"
    >
      delete
    </i>
  </a>
  <a
    className="secondary-content"
    href="!#"
    onClick={[Function]}
  >
    <i
      className="material-icons orange-text"
    >
      mode_edit
    </i>
  </a>
</li>
`;
